staging/src/k8s.io/apimachinery/pkg/api/resource/quantity_test.go

```diff
@@ -1137,6 +1137,42 @@ func TestAdd(t *testing.T) {
 	}
 }
 
+func TestMul(t *testing.T) {
+	tests := []struct {
+		a        Quantity
+		b        int64
+		expected Quantity
+		ok       bool
+	}{
+		{decQuantity(10, 0, DecimalSI), 10, decQuantity(100, 0, DecimalSI), true},
+		{decQuantity(10, 0, DecimalSI), 1, decQuantity(10, 0, DecimalSI), true},
+		{decQuantity(10, 0, BinarySI), 1, decQuantity(10, 0, BinarySI), true},
+		{Quantity{Format: DecimalSI}, 50, decQuantity(0, 0, DecimalSI), true},
+		{decQuantity(50, 0, DecimalSI), 0, decQuantity(0, 0, DecimalSI), true},
+		{Quantity{Format: DecimalSI}, 0, decQuantity(0, 0, DecimalSI), true},
+
+		{decQuantity(10, 0, DecimalSI), -10, decQuantity(-100, 0, DecimalSI), true},
+		{decQuantity(-10, 0, DecimalSI), 1, decQuantity(-10, 0, DecimalSI), true},
+		{decQuantity(10, 0, BinarySI), -1, decQuantity(-10, 0, BinarySI), true},
+		{decQuantity(-50, 0, DecimalSI), 0, decQuantity(0, 0, DecimalSI), true},
+		{decQuantity(-50, 0, DecimalSI), -50, decQuantity(2500, 0, DecimalSI), true},
+		{Quantity{Format: DecimalSI}, -50, decQuantity(0, 0, DecimalSI), true},
+
+		{decQuantity(mostPositive, 0, DecimalSI), 10, decQuantity(mostPositive, 1, DecimalSI), false},
```

Comment:
I'd like all these covered:

```
	{decQuantity(mostPositive, 0, DecimalSI), 0, decQuantity(0, 1, DecimalSI), true},
	{decQuantity(mostPositive, 0, DecimalSI), 1, decQuantity(mostPositive, 1, DecimalSI), true},
	{decQuantity(mostPositive, 0, DecimalSI), -1, decQuantity(-mostPositive, 1, DecimalSI), true},
	{decQuantity(mostPositive/2, 0, DecimalSI), 2, decQuantity(mostPositive, 1, DecimalSI), true},
	{decQuantity(mostPositive/-2, 0, DecimalSI), -2, decQuantity(mostPositive, 1, DecimalSI), true},
	{decQuantity(mostPositive, 0, DecimalSI), 2, decQuantity(mostPositive, 1, DecimalSI), false},

	{decQuantity(mostNegative, 0, DecimalSI), 0, decQuantity(0, 1, DecimalSI), true},
	{decQuantity(mostNegative, 0, DecimalSI), 1, decQuantity(mostNegative, 1, DecimalSI), true},
	{decQuantity(mostNegative, 0, DecimalSI), -1, decQuantity(mostNegative, 1, DecimalSI), false},
	{decQuantity(mostNegative/2, 0, DecimalSI), 2, decQuantity(mostNegative, 1, DecimalSI), true},
	{decQuantity(mostNegative/-2, 0, DecimalSI), -2, decQuantity(mostNegative, 1, DecimalSI), false},
	{decQuantity(mostNegative, 0, DecimalSI), 2, decQuantity(mostNegative, 1, DecimalSI), false},
```

Reply:
It makes sense! Thanks for the great suggestions!

Reply:
done.